libsplinter/src/biome/rest_api/actix/verify.rs

```diff
@@ -0,0 +1,126 @@
+                            debug!("Failed to fetch credentials: {}", err);
+                            match err {
+                                CredentialsStoreError::NotFoundError(_) => {
+                                    return HttpResponse::BadRequest()
+                                        .json(ErrorResponse::bad_request(&format!(
+                                            "Username not found: {}",
+                                            username_password.username
+                                        )))
+                                        .into_future()
+                                }
+                                _ => {
+                                    error!("Failed to fetch credentials: {}", err);
+                                    return HttpResponse::InternalServerError()
+                                        .json(ErrorResponse::internal_error())
+                                        .into_future()
+                                }
+                            }
+                        }
+                    };
+
+                match authorize_user(&request, &credentials.user_id, &secret_manager, &rest_config) {
+                    AuthorizationResult::Authorized => {
+                        match credentials.verify_password(&username_password.hashed_password) {
+                            Ok(true) => {
+                                    HttpResponse::Ok()
+                                        .json(json!({ "message": "Successful verification", "user_id": credentials.user_id }))
+                                        .into_future()
+                                    }
+                            Ok(false) => {
+                                HttpResponse::BadRequest()
+                                    .json(ErrorResponse::bad_request("Invalid password"))
+                                    .into_future()
+                            }
+                            Err(err) => {
+                                error!("Failed to verify password: {}", err);
+                                HttpResponse::InternalServerError()
+                                    .json(ErrorResponse::internal_error())
+                                    .into_future()
+                            }
+                        }
+                    }
+                    AuthorizationResult::Unauthorized(msg) => {
+                        HttpResponse::Unauthorized()
+                                .json(ErrorResponse::unauthorized(&msg))
+                                .into_future()
+                    }
+                    AuthorizationResult::Failed => {
+                        error!("Failed to authorize user");
+                            HttpResponse::InternalServerError()
+                                .json(ErrorResponse::internal_error())
+                                .into_future()
+                    }
+                }
+            }))
+        })
+}
```

Comment:
Please add some tests if possible; see the the circuit/proposal read tests for examples of how to test endpoints: https://github.com/Cargill/splinter/blob/master/libsplinter/src/admin/rest_api/actix/circuits.rs

Reply:
As it stands, there is a lot of background work that needs to go into setting up the unit tests. It involved editing the existing *-stores and how they're passed around to make the biome resources. There is a story to add in these tests: https://zenhub.cglcloud.com/app/workspaces/product-core-5d37298b0add7306b30efde2/issues/digitalsupplychain/product-core/2019